plans/example-browser/failure.js

```diff
@@ -0,0 +1,11 @@
+module.exports = async (page, runenv, client) => {
+  const userAgent = await page.evaluate(() => {
+    return window.navigator.userAgent
+  })
+  if (userAgent !== 'testground') {
+    throw new Error(`expected user agent 'testground', unexpected: ${userAgent}`)
+  }
+
+  // we do not ever expect to reach here, this is a failure test
+  runenv.recordMessage('playwright: success: testground platform detected')
+}
```

Comment:
I would tweak the Error / success messages to signify that the expected behavior is for this test to fail, and not reach the end of the function.

Reply:
If the `failure.js` is integrated into the CI, or if someone runs the test on their own, the messages can be a bit confusing. Here's what I would do: the default output should be something like 
`failure.js: If you see this message, that means the test failed as expected. User agent is ${}.`
And for the failure
`failure.js: If you see this message, something bad has occured - this test is meant to fail`

That way, if I see the message `failure.js - PASSED X` (or similar), it won't be as confusing.
Thoughts?

Reply:
Let's simplify and use the same code that we used in `failure.js`
it's explicit and self-documented.
https://github.com/testground/testground/blob/62c18ac2150aae5f49a091b03150872440e8c564/plans/example-js/failure.js

Reply:
Went with @brdji's proposal, seems the most fair IMHO.